parsing/pcoq.ml

```diff
@@ -587,12 +587,31 @@ let recover_grammar_command (type a) (tag : a grammar_command) : a list =
 
 let extend_dyn_grammar (GrammarCommand.Dyn (tag, g)) = extend_grammar_command tag g
 
+(* Tactic parsing modes *)
+let add_proof_tactic_entry p e = Hashtbl.add tactic_entry p e
```

Comment:
That was going to be my question: so you want to switch proof mode while parsing a sentence? I understand the motivations, but *timeo CamlpX et dona ferentem*. If there is a part of Coq I deeply mistrust, this is parsing indeed. Probably after unification, modules, universes, and a few more components. But still, I'm concerned by the fragility and the scalability of such a feature.